src/supertux/menu/addon_gallery.cpp

```diff
@@ -0,0 +1,182 @@
+//  SuperTux
+//  Copyright (C) 2018 christ2go <christian@hagemeier.ch>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "physfs/physfs_file_system.hpp"
+#include "supertux/menu/addon_gallery.hpp"
+#include "util/file_system.hpp"
+#include <physfs.h>
+#include <set>
+#include <string>
+
+std::string addon_type_to_translated_string(Addon::Type type) {
+  switch (type) {
+    case Addon::LEVELSET:
+      return _("Levelset");
+
+    case Addon::WORLDMAP:
+      return _("Worldmap");
+
+    case Addon::WORLD:
+      return _("World");
+
+    case Addon::LANGUAGEPACK:
+      return "";
+
+    default:
+      return _("Unknown");
+  }
+}
+
+void AddonGallery::refresh() {
+  clear();
+  Downloader d;
+  // If the Add-On is installed do not download, instead load locally
+  try {
+    std::vector<std::string> installedAddons =
+        m_addon_manager.get_installed_addons();
+    bool installed = (std::find(installedAddons.begin(), installedAddons.end(),
+                                m_addon) != installedAddons.end());
+    Addon &a = installed ? m_addon_manager.get_installed_addon(m_addon)
+                         : m_addon_manager.get_repository_addon(m_addon);
+    std::vector<std::pair<std::string, std::string>> availableScreenshots;
+    // If installed mount the add-on
+    std::string mountpoint;
```

Comment:
This was copied from https://github.com/SuperTux/supertux/blob/f6fa35439a74f141a97614e56e3be4db44005a03/src/addon/addon_manager.cpp#L447. 

Can you move this into a new method inside addon_manager please?